client/ayon_houdini/api/hda_utils.py

```diff
@@ -808,7 +813,9 @@ def get_available_representations(node):
     representations = get_representations(
             project_name,
             version_ids={version_entity["id"]},
-            fields={"name"}
+            fields={"name"},
+            representation_names=ALLOWED_USD_REPRESENTATIONS \
+                if node.parm("usd_compatible_representations") and node.parm("usd_compatible_representations").eval() else None
```

Comment:
Why is this `usd` specific?

Also, by adding a default filter here it basically avoids the generic loader being useful **everywhere**. It should not filter by default, unless e.g. something on the node specifies a filter?

Reply:
This toggle only found on `Generic Loader` lop node. 
and, I had to provide such a filter because the Asset/Shot load node will then have representations it can't load like `exr` or `vdb`.  
![image](https://github.com/user-attachments/assets/ec156bc5-f14f-413d-aed1-8d019578dd31)

Reply:
Instead of doing that - add a built-in feature for any generic loader to specify a (unique) filter  by having a hidden parm on the generic loader that takes a space separated list of representation names to filter to- when empty no filtering is applied. By default, it's empty.

Then your node can just set that parm to e.g. `abc usd` on the node itself

Reply:
https://github.com/ynput/ayon-houdini/pull/199/commits/714fe325f5aad3189cf82a42c1f50e4084dc8e90